options/ansi/generic/time-stubs.cpp

```diff
@@ -105,164 +107,232 @@ size_t strftime(char *__restrict dest, size_t max_size,
-		}else if (*(c + 1) == 'F') {
-			auto chunk = snprintf(p, space, "%d/%.2d/%.2d", 1900 + tm->tm_year, tm->tm_mon + 1,
+			c++;
+			break;
+		}
+		case 'F': {
+			chunk = snprintf(p, space, "%d-%.2d-%.2d", 1900 + tm->tm_year, tm->tm_mon + 1,
 					tm->tm_mday);
 			if(chunk >= space)
 				return 0;
 			p += chunk;
-			c += 2;
-		}else if (*(c + 1) == 'a') {
-			const char *strdays[] = { "Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat" };
+			c++;
+			break;
+		}
+		case 'D': {
+			chunk = snprintf(p, space, "%.2d/%.2d/%.2d", tm->tm_mon + 1, tm->tm_mday, tm->tm_year % 100);
+			if(chunk >= space)
+				return 0;
+			p += chunk;
+			c++;
+			break;
+		}
+		case 'a': {
 			int day = tm->tm_wday;
 			if(day < 0 || day > 6)
 				__ensure(!"Day not in bounds.");
 
-			auto chunk = snprintf(p, space, "%s", strdays[day]);
+			chunk = snprintf(p, space, "%s", nl_langinfo(DAY_1 + day));
 			if(chunk >= space)
 				return 0;
 			p += chunk;
-			c += 2;
-		}else if (*(c + 1) == 'b' || *(c + 1) == 'B') {
-			const char *strmons[12];
-			switch (*(c + 1)) {
-				case 'b':
-					strmons[0] = "Jan";
-					strmons[1] = "Feb";
-					strmons[2] = "Mar";
-					strmons[3] = "Apr";
-					strmons[4] = "May";
-					strmons[5] = "Jun";
-					strmons[6] = "Jul";
-					strmons[7] = "Aug";
-					strmons[8] = "Sep";
-					strmons[9] = "Oct";
-					strmons[10] = "Nov";
-					strmons[11] = "Dec";
-					break;
-				case 'B':
-					strmons[0] = "January";
-					strmons[1] = "February";
-					strmons[2] = "March";
-					strmons[3] = "April";
-					strmons[4] = "May";
-					strmons[5] = "June";
-					strmons[6] = "July";
-					strmons[7] = "August";
-					strmons[8] = "September";
-					strmons[9] = "October";
-					strmons[10] = "November";
-					strmons[11] = "December";
-					break;
-			}
+			c++;
+			break;
+		}
+		case 'b':
+		case 'B':
+		case 'h': {
 			int mon = tm->tm_mon;
 			if(mon < 0 || mon > 11)
 				__ensure(!"Month not in bounds.");
 
-			auto chunk = snprintf(p, space, "%s", strmons[mon]);
+			nl_item item = (*c == 'B') ? MON_1 : ABMON_1;
+
+			chunk = snprintf(p, space, "%s", nl_langinfo(item + mon));
 			if(chunk >= space)
 				return 0;
 			p += chunk;
-			c += 2;
-		}else if (*(c + 1) == 'c') {
-			auto chunk = snprintf(p, space, "%d/%.2d/%.2d %.2d:%.2d:%.2d", 1900 + tm->tm_year,
+			c++;
+			break;
+		}
+		case 'c': {
+			chunk = snprintf(p, space, "%d/%.2d/%.2d %.2d:%.2d:%.2d", 1900 + tm->tm_year,
 					tm->tm_mon + 1, tm->tm_mday, tm->tm_hour, tm->tm_min, tm->tm_sec);
 			if(chunk >= space)
 				return 0;
 			p += chunk;
-			c += 2;
-		}else if (*(c + 1) == 'd') {
-			auto chunk = snprintf(p, space, "%.2d", tm->tm_mday);
-			if(chunk >= space)
-				return 0;
-			p += chunk;
-			c += 2;
-		}else if (*(c + 1) == 'e') {
-			int chunk;
+			c++;
+			break;
+		}
+		case 'e': {
 			chunk = snprintf(p, space, "%2d", tm->tm_mday);
 			if(chunk >= space)
 				return 0;
 			p += chunk;
-			c += 2;
-		}else if (*(c + 1) == 'I') {
+			c++;
+			break;
+		}
+		case 'I': {
```

Comment:
`tm_hour` can be 0, can't it? In which case our logic here (and in `%r`) is wrong, since it needs to be printed in the range `01-12`.